finality-aleph/src/sync/ticker.rs

```diff
@@ -0,0 +1,115 @@
+use tokio::time::{sleep, Duration, Instant};
+
+/// This struct is used for rate limiting as an on-demand ticker. It can be used for ticking
+/// at least once `max_timeout` but not more than once every `min_timeout`.
+pub struct Ticker {
```

Comment:
I wonder about the usage of this. The tick rate depends on how often the calls are made and is not guaranteed to tick once every `max_timetout`. My suggestion is to either change docs to describe this or slightly refactor this struct:
1. fn run that runs ticker and sends a notification every max_timeout
2. fn try_tick that tries to tick immediately and sends a notification
It would probably be required to split Ticker into Ticker (the interface with which the caller interacts) and TickerWorker doing all of the work

Reply:
This proposal doesn't actually improve things, you just move the problem from awaiting on the method to awaiting on the channel. Worse, it breaks the main guarantee we are interested in, that no ticks happen sooner than `min_timeout`, or at least I don't see how you can uphold it with a channel.

Not to mention it's massively more complicated. xd

Reply:
It makes it more complicated here but easier to use for the user of the struct. Anyway, I'm fine here with just docs reflecting what I wrote in the first paragraph :)

Reply:
I think that approach with ticker worker is unnecessary complicated and current usage for user is pretty simple. You just add `wait` to your main select loop and use `try_tick` whenever you would like to tick sooner in another branch of select. So in this case what I wrote in docs is guaranteed (apart from some stupid cases with heavy synchronous tasks). If you want to I can add this to the doc.

Reply:
Yes, but it is not guaranteed by the ticker itself